app/controllers/articles_controller.rb

```diff
@@ -1,51 +1,85 @@
 class ArticlesController < ApplicationController
 
-  http_basic_authenticate_with name: "dhh", password: "secret", 
-  except: [:index, :show]
+  before_action :logged_in_user
 
-  def index
-    @articles = Article.all
+  def home
   end
 
-  def show
-    @article = Article.find(params[:id]) #Instance variable
+  def new
+    @article = current_user.articles.new
   end
   
-  def new
-    @article = Article.new
+  def index
+    @articles = Article.all.select {|a| a.status == 'public'}
+
   end
 
+  def user_index
+    # if current_user?(params[:id].to_i)
+    @articles = Article.all
+    @articles = @articles.select { |a| a.user_id == @current_user.id }
+    render 'user_index'
+    # else
+    #   redirect_to "/user/#{ current_user.id }"
+    # end
+
+  end
+    
   def create
-    @article = Article.new(article_params)
+    @article = current_user.articles.build(article_params)
 
     if @article.save
+      flash[:success] = 'Article haas been created!'
       redirect_to @article
     else
       render :new 
     end
   end
 
   def edit
-    @article = Article.find(params[:id])
+    @article = current_user.articles.find(params[:id])
   end
 
   def update
-    @article = Article.find(params[:id])
+    @article = current_user.articles.find(params[:id])
 
     if @article.update(article_params)
+      flash[:success] = 'Article updated'
       redirect_to @article
     else
       render :edit
     end
   end
   
   def destroy
-    @article = Article.find(params[:id])
-    @article.destroy
+    @article = current_user.articles.find(params[:id])
+    if @article
+      @article.destroy
+      flash[:success] = 'Article has been deleted'
+    else
+      flash[:alert] = 'Error'
+    end
     redirect_to root_path
   end
+
+  def show
+    @article = Article.find(params[:id]) #Instance variable
+    puts "ID DEL USER #{ @current_user.id} ID del ARTICULO #{ @article.user_id}"
+    puts "ART de #{@article.title}, loggeado #{@current_user.id.class}" 
+    puts "Iguales? #{@article.user_id == @current_user.id}"
```

Comment:
are these `puts` used for debugging? in case yes please delete them, if not change them to English